test/new-e2e/utils/infra/stack_manager.go

```diff
@@ -60,8 +60,24 @@ func newStackManager(ctx context.Context) (*StackManager, error) {
 	}, nil
 }
 
+func getNewStack(ctx context.Context, name, project string, deployFunc pulumi.RunFunc, failOnMissing bool) (auto.Stack, error) {
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
It can be confusing to fail on non existent when calling `getNewStack`: if it is new, I expect it to be missing. This check could be implemented directly in `GetStack`: if `failOnMissing` is true we call `auto.SelectStackInlineSource`, otherwise we call `auto.UpsertStackInlineSource`. Hiding those calls behind `getNewStack` can be confusing.